src/main/java/io/cloudtrust/testclient/TokenInfoController.java

```diff
@@ -132,11 +183,11 @@ private String buildTokenInfo(Principal p, HttpSession session) {
                     String assertionFromProfile = (String) profile.getAttribute("saml_assertion");
                     if (assertionFromSession != null) {
                         // assertion from the session
-                        assertionStr = formatXML(assertionFromSession);
+                        assertionStr = formatXML(new String(Base64.getDecoder().decode(assertionFromSession),StandardCharsets.UTF_8));
```

Comment:
Still compatible with legacy code?

Reply:
I've tested the login with artifact binding and it still works. So I'd say it is compatible.
Do you have a specific use case in mind?